src/VecSim/vec_sim_common.h

```diff
@@ -63,12 +63,14 @@ typedef enum {
     VecSimParamResolverErr_InvalidPolicy_AdHoc_With_EfRuntime
 } VecSimResolveCode;
 
+typedef struct AsyncJob AsyncJob; // forward declaration.
+
 /**
  * Callback signatures for asynchronous tiered index.
  */
-typedef int (*SubmitCB)(void *job_queue, void **jobs, size_t jobs_len);
+typedef int (*SubmitCB)(void *job_queue, AsyncJob **jobs, size_t jobs_len, void *index_ctx);
```

Comment:
I don't think `AsyncJob` is a well known type outside of the library.

Reply:
I moved it to `vec_sim_common.h` (as opaque object) so it will be known outside as well